src/cpp/session/modules/clang/RCompilationDatabase.cpp

```diff
@@ -349,7 +349,25 @@ std::string computeCompilerHash(bool isCpp)
    
    std::stringstream ss;
    ss << std::hash<std::string>{}(string_utils::trimWhitespace(result.stdOut));
-   return ss.str();
+   std::string hash = ss.str();
+   
+   return hash;
+}
+
+// NOTE: We assume that the compiler will not change within a single R session.
+// If the user wants to change the compiler they're using, they should restart R.
+std::string computeCompilerHash(bool isCpp)
+{
+   if (isCpp)
```

Comment:
Had a momentary double-take with defining a same-named static variable in separate scopes inside a function. Valid, and they are distinct variables (e.g. memory locations). This is really a note to myself, not you. :-)